tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -0,0 +1,337 @@
+                        for (int i = 0; i < MAX_BATCH_SIZE; i++) {
+                            String topic = topicsToCreate.pollFirst();
+                            topicsToCreate.remove(topic);
+                            newTopics.add(new NewTopic(topic, 1, REPLICATION_FACTOR));
+                        }
+                        futures.add(adminClient.createTopics(newTopics).all());
+                    }
+                    for (Future<Void> future : futures) {
+                        future.get();
+                    }
+                    log.info("Successfully created " + spec.totalTopics() + " topics.");
+                }
+                executor.submit(new SendRecords());
+            } catch (Exception e) {
+                abort("CreateBenchmarkTopics", e);
+            }
+            return null;
+        }
+    }
+
+    private static class SendRecordsCallback implements Callback {
+        private final SendRecords sendRecords;
+        private final long startMs;
+
+        SendRecordsCallback(SendRecords sendRecords, long startMs) {
+            this.sendRecords = sendRecords;
+            this.startMs = startMs;
+        }
+
+        @Override
+        public void onCompletion(RecordMetadata metadata, Exception exception) {
+            long now = Time.SYSTEM.milliseconds();
+            long durationMs = now - startMs;
+            sendRecords.recordDuration(durationMs);
+        }
+    }
+
+    public class SendRecords implements Callable<Void> {
+        private final Histogram histogram;
+
+        private final Throttle throttle;
+
+        private final Future<?> statusUpdaterFuture;
+
+        SendRecords() {
+            this.histogram = new Histogram(5000);
+            this.throttle = new Throttle(spec.targetMessagesPerSec());
+            this.statusUpdaterFuture = executor.scheduleWithFixedDelay(
+                new StatusUpdater(histogram), 1, 1, TimeUnit.MINUTES);
+        }
+
+        @Override
+        public Void call() throws Exception {
+            long startTimeMs = Time.SYSTEM.milliseconds();
+            try {
+                Properties props = new Properties();
+                props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
+//                props.put(ProducerConfig.BATCH_SIZE_CONFIG, 16384);
+//                props.put(ProducerConfig.ACKS_CONFIG, 1);
+//                props.put(ProducerConfig.LINGER_MS_CONFIG, 0);
+//                props.put(ProducerConfig.SEND_BUFFER_CONFIG, 128 * 1024);
+//                props.put(ProducerConfig.RECONNECT_BACKOFF_MS_CONFIG, 50);
+//                props.put(ProducerConfig.BUFFER_MEMORY_CONFIG, 32 * 1024 * 1024L);
+                for (Map.Entry<String, String> entry : spec.producerConf().entrySet()) {
+                    props.setProperty(entry.getKey(), entry.getValue());
+                }
+                byte[] key = new byte[512];
+                byte[] value = new byte[512];
```

Comment:
Are the sizes not configurable? The constants are too hidden here, it may be better to declare them as a static at the start of the class if not configurable.

Reply:
I'm going to make this configurable later (along with a bunch of other stuff) but for now, yeah, let's have a constant.